server/graphql/v2/mutation/CreateCollectiveMutation.js

```diff
@@ -22,96 +24,119 @@ async function createCollective(_, args, req) {
-  if (args.automateApprovalWithGithub && args.collective.githubHandle) {
-    const githubHandle = args.collective.githubHandle;
-    const opensourceHost = defaultHostCollective('opensource');
-    host = await loaders.Collective.byId.load(opensourceHost.CollectiveId);
-    try {
-      const githubAccount = await models.ConnectedAccount.findOne({
-        where: { CollectiveId: remoteUser.CollectiveId, service: 'github' },
-      });
-      if (!githubAccount) {
-        throw new Error('You must have a connected GitHub Account to create a collective with GitHub.');
+  let user = remoteUser,
+    host;
+
+  return sequelize
+    .transaction(async transaction => {
+      if (!user && args.user && args.host.legacyId === defaultHostCollective('foundation').CollectiveId) {
+        user = await models.User.findByEmail(args.user.email, transaction);
+        if (!user) {
+          user = await models.User.createUserWithCollective(args.user, transaction);
+        } else {
+          throw new Unauthorized('Email already exist, you have to be logged in to apply with this email');
+        }
+      } else if (!user) {
+        throw new Unauthorized('You need to be logged in to create a collective');
       }
-      // In e2e/CI environment, checkGithubAdmin and checkGithubStars will be stubbed
-      await github.checkGithubAdmin(githubHandle, githubAccount.token);
-      await github.checkGithubStars(githubHandle, githubAccount.token);
-      shouldAutomaticallyApprove = true;
-    } catch (error) {
-      throw new ValidationFailed(error.message);
-    }
-    if (githubHandle.includes('/')) {
-      collectiveData.settings.githubRepo = githubHandle;
-    } else {
-      collectiveData.settings.githubOrg = githubHandle;
-    }
-    collectiveData.tags = collectiveData.tags || [];
-    if (!collectiveData.tags.includes('open source')) {
-      collectiveData.tags.push('open source');
-    }
-  } else if (args.host) {
-    host = await fetchAccountWithReference(args.host, { loaders });
-    if (!host) {
-      throw new ValidationFailed('Host Not Found');
-    }
-    if (!host.isHostAccount) {
-      throw new ValidationFailed('Host account is not activated as Host.');
-    }
-  }
-
-  const collective = await models.Collective.create(collectiveData);
 
-  // Add authenticated user as an admin
-  await collective.addUserWithRole(remoteUser, roles.ADMIN, { CreatedByUserId: remoteUser.id });
-
-  // Add the host if any
-  if (host) {
-    await collective.addHost(host, remoteUser, { shouldAutomaticallyApprove, message: args.message });
-    purgeCacheForCollective(host.slug);
-  }
+      const collectiveData = {
+        slug: args.collective.slug.toLowerCase(),
+        ...pick(args.collective, ['name', 'description', 'tags']),
+        isActive: false,
+        CreatedByUserId: user.id,
+        settings: { ...DEFAULT_COLLECTIVE_SETTINGS, ...args.collective.settings },
+        ...(args.applicationData && { data: args.applicationData }),
```

Comment:
As per https://github.com/opencollective/opencollective-api/pull/5075#discussion_r552475720, we want to store the application data in the `HostApplication`, not the collective `data`. We should pass `applicationData` to `addHost` in https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v2/mutation/CreateCollectiveMutation.js#L92 then pass it down to `recordApplication` through the `customData` key in https://github.com/opencollective/opencollective-api/blob/master/server/models/Collective.js#L1929.